drivers/can/can.c

```diff
@@ -556,7 +556,7 @@ static ssize_t can_read(FAR struct file *filep, FAR char *buffer,
   FAR struct can_dev_s     *dev = inode->i_private;
 #endif
 
-  caninfo("buflen: %d\n", buflen);
+  caninfo("buflen: %d\n", (int)buflen);
```

Comment:
Put these changes to another PR and upstream it.

Also, size_t is unsigned int, so the correct way to do this would be

caninfo("buflen: %lu\n", (unsigned long)buflen);

Unsigned long guarantees that whatever sized size_t is, the print works.

Reply:
C99 is commonly used nowdays and inttypes.h. So you can just change "%d" into "%zu" IMHO

Reply:
Won't work with NuttX base code, which is ANSI C